### ISSUE TITLE
Support tf.saved_model to save and restore trained models in the SavedModel format

### ISSUE THREAD

**Issue by ashbhandare** [SciSharp/TensorFlow.NET#445]
Current support is for tf.train.saver, which saves checkpoint files. The saved_model format is better for deploying trained models according to https://www.tensorflow.org/guide/saved_model.

**Comment by Oceania2018** [comment]
@ashbhandare Are you talking about SavedModel Warmup: https://www.tensorflow.org/tfx/serving/saved_model_warmup ?

**Comment by ashbhandare** [comment]
I meant https://www.tensorflow.org/api_docs/python/tf/saved_model/save in general, but https://www.tensorflow.org/tfx/serving/saved_model_warmup looks like what we were looking for, for our specific issue. Doesn't the saved_model_warmup require saving the model in the saved_model format?

**Comment by PleasantD** [comment]
I'm looking for this so that I can use the TFLiteConverter.
Alternatively, I need support of {{tf.quantization.fake_quant_with_min_max_args}}, I think.

**Comment by eduardobellotto** [comment]
Google Cloud Platform AutoML saves models just in this new format: saved_model.pb

**Comment by m0nzderr** [comment]
Being able to load and call signatures from generic tf.saved_model format would be very useful!
